rfcs/experimental/flow_graph_wait_try_put_and_wait/README.md

```diff
@@ -0,0 +1,604 @@
+associated with the tuple and submitted to successors.
+
+Similar to the ``queueing`` case, if one of the input ports was triggered with the input value, but others never receive any values, the item and the metainformation would be kept in the
+buffer and block the corresponding ``try_put_and_wait`` call.
+
+### Key-matching ``join_node``
+
+Behaves the same as other ``join_node`` policies, except the values and the metainformation objects are stored in the hash map inside of the port.
+
+### ``split_node``
+
+The split node takes the tuple object and the corresponding metainformation, split the tuple and submit the single values from the tuple to the corresponding ports. 
+Metainformation object copy is submitted together with each element into each output port.
+
+Metainfo should not be split since is is unclear what is the relation between elements in the tuple and the metainformation objects stored in the internal list so all of them should
+be provided to the successors.
+
+### ``indexer_node``
+
+``indexer_node`` only tags the input value and broadcasts it with the original metainfo to the successors.
+
+### ``composite_node``
+
+``composite_node`` does not require any additional changes. `try_put_and_wait` feature and the explicit support for metainformation should be done by the nodes inside of the composite.
+
+## Additional implementation details
+
+### Details about metainformation class
+
+``message_metainfo`` class synopsis:
+
+```cpp
+class message_metainfo
+{
+public:
+    using waiters_type = std::forward_list<d1::wait_context_vertex*>;
+
+    message_metainfo();
+
+    message_metainfo(const message_metainfo&);
+    message_metainfo(message_metainfo&&);
+
+    message_metainfo& operator=(const message_metainfo&);
+    message_metainfo& operator=(message_metainfo&&);
+
+    const waiters_type& waiters() const &;
+    waiters_type&& waiters() &&;
+
+    bool empty() const;
+
+    void merge(const message_metainfo&);
+};
+```
+
+The current implementation of ``message_metainfo`` class wraps only the list of single message waiters. The class may be extended if necessary to cover additional use-cases.
+
+The metainfo is required to hold a list of message waiters instead of single waiter to cover the ``continue_node`` and ``join_node`` joining use-cases. Consider the example:
+
+```cpp
+using namespace oneapi::tbb;
+
+flow::function_node<int, int> start1(g, ...);
+flow::function_node<int, int> start2(g, ...);
+
+flow::join_node<std::tuple<int, int>> join(g);
+
+flow::function_node<std::tuple<int, int>, int> post_process(g, ...);
+
+flow::make_edge(start, flow::input_port<0>(join));
+flow::make_edge(start, flow::input_port<1>(join));
+flow::make_edge(join, post_process);
+
+std::thread t1([&]() {
+    start1.try_put_and_wait(1);
+});
+
+std::thread t2([&]() {
+    start2.try_put_and_wait(2);
+})
+```
+
+### Combined or separated wait
```

Comment:
It looks like this question is not added to the "Exit criteria" list; should it be?